docs/snippets/azkv-pushsecret-certificate.yaml

```diff
@@ -0,0 +1,26 @@
iCmAlHW/+5RyYXUL/c1bvlBMZz6z9n3lXBjnee3kRVbrQV/aa+069Nd76Mi8WBXhsjkm3+K14fuJksg1x3NohcL3/kW/iYHBXBFug2w6wX8l1T3XcxekBKDDNDfoy/ZExmDmsDAmUfIWb4zonLlaGq6z/9l3LkjrNS9/7C2YYEmh0xmMGD0mjzj1k932LWzvLdPaEIZTm2YsoI4TyraKc2yjXmmQWldGeN4yB/s0RfoaxPrMjkQsO6ZaU+gPqucd0JjL/e9gDVUWB0CL6Wy1T+H60iWBQhj0h5PoXuIbsocVj9PC08zniXmXUiVvbt62AyPm7oYci8BXJIc18WQ/BcO1EzPbQCVqI8DMWkyVCN/XUR7Oufseib//qZyf5XOSqwTHaetWU5tsIVKHcU8vzwPHqiDrFRmkBvn8pmtaQibaartr1XmiTCCkRX5yCzstbWWhjnRB/UD1zTxiMQUEmkmFD3zldLYuXWaaQNpNExog7vyJSMLUuTpff68sElBTRUA5pqqiVua5VsorzWaVMORsdagii4iJ/KQDoSs3KLZ6tXqNHBxoP69Mf6WJF8LZ+3FyYr+Ckpii2zgxx7D274fK6XfQpTS1yxeJPTxPZoEEZs9G03UKkrfi3Uv106akNB4XyAMxsyK26GTbKGkSig1C2gme/oCJfG8ZYin8hhl6QTlGk4ao2RWXT8Rh+qW6PDm+SVA4WJl9NI/eJwEtjmE2U1nt0IfkNy2miug3rHbr1yso4gnfdQUtMsHz055y7GgauJKADGa7N2dUdq7jbK4rbD59QsloYKpkN0TX2g86APMuMlHai33A942bsFnp+IC4ONHpj+LQZHgbr1ygVhI6EQs+x6OAw/0UrEq8fP0KVJBZHR96GJZHGCkVy6z+FVlRGTw4Lgb+dMpjucJeLQw852TLN3zHKsCglsygzGf6dfZhHs4o0tB+NyWr4Pgw/F8n3iXhTQCv95fHoTz9lav/LU1KTxADPhgczNUE6e2zrfZ0724d+eBOZXPY5endc+t2Kci+O2S2xS3XDxt4GDEWEogy6kjQoObjaHEoiOUcplbTfgptpXseivK1y3DU3gJPbNYTE8qJIOnFDwUxYDGYjlUH9h+SguCeRkO8sR5iQrYsjQQbDcPxss3rg6DeytcnsFwKf8bE/mwOpzHOE9CMrWuOoPrmtn7Lad7aTn1YN3hyeaHHVVAWzsbOyhhvEmsX5pifM26ZF57cabEiOU3itt5cYI9qxb90V4hc69NgfONx89RtrgsXd61rIsf8CeQDx3rQmQWZ54Pb0o5WUMnCbui/Z5Nw4RjeKdl38josYB3VZCXBz7g1y3ZWZUpl8GTn6TUZGXtXw8e4g18RJxpgt5OC3d27jY4fsDrVlUlYuZV21fbOG/MyT1YKQvNZlazpNPXXXyHK+Swj+A3H4It0K7IhM4R5+riH3ngcgbtF3M3StIM4lT5KVA84CRUWmDPIUskNbTUJWLw3nhDa5mmgCC9eRrb724leAFWG/FAvlsXmMl5BWWm5KYynu+y0pNpDbLmNNP6PlEBi9Ipms51BaUbQa2Y+LbFXj7dm4MBl07qroxQTOAu/DmUrxkBXU0ElpYhYrYoGbODuH/fD0C2KOJ5d6O8jRg3nN0228A2wrd18p47Q63mHyJlezxKI69o1Um6FNUjTtk6KU1Zp0AtW4mE5jkdKLfRNhlDPrgZAeYoT3OFU9UaPRauw/EJBNhSjmZC6aa4wqu60gmQ8xfnJ6YEd/XayZXqjoAiE9cAC3tF3bO8x3gpp0D0uKFhha5VgtYb9LTVlbE+Kb4DLi5fEkRWyTfywVMXLPgm5HnVE3Pz6SRwH8jHUjNSnqRRbwJfJ4Sb8LzQSzf1y0imoAfGRUcjbIC0tKAitYZ5LduaIzNyi4hBaAwsXLkLEkBClUqtI4LtCf6ETgPNRWmSWYKeiV6vzjyR0xhBi1VixpS8HN4OdA8Te3t41lp6w8nFaJcrD0LnLNPMeotJv5u+6gc3d1Jd/to+DtvxpFulh8J5WMEXCHJOV6nSKRb+rGUFh+2jp9qkSB98XGCcKn3blAFxEGYwtJXqpP53pPxa6Up/f0/KR87a4IWD0y5fZ90+HcvDCnQofuQWsMobpiFWN7ScYttbcO85xv7J6Qs+yUf9hzraN8rLW9o7LKnW+fCT8c0ggTIqTyaW1HwTtYDZtYDUIG9KDj3KY1YZN2yca+ErnS1phfI71m8JqjvH2k/+finxs2IexESmjzRTeqgEKGrhx1BdhwJ5/nd9moF6HuWqR3XCd2UtiLyhQbFt+Lo70vHm3m0Duzu03pKgGGXQQQqa1QM6YIAV6s3TSs/5cYm5KGZCd7UoKlMRzivHst/Nm6Zy+3jckpn626f7RkP/hIfT9Qvrd80PGYeCe2nNvTMfAfScYbczZ225knqeT6Z7vdY8+jgabAjREPLzKvZlL3wS6FquFEsGYn/BNAbkuJ9OhxWBgwqhTVLGaJdTAtl9cgcJqUKE76cclwit9ZF0ucUBdqV9twE2prGB/ujSKmhJ7Qd/FwEr7/UdJwQ8iVbs3+qJHBg91WPhTR9eab0YHzM+62FePZFWpQQ8m9RfP5Ku262YLhGEqmBHAHcOomhWUF/t3fQWewUIADG1Mr4nICeYUbLjsnS3IpASwM2uzFNBgdIe/i/xq5KZMvjtaaEqUviVPkAcHrS96L58DoEnsC+96ljH9lyBwJcJ3q2eT7rQNFY/yANRvNi5ix1mtZV8J6d/HWr2v5P67W3TnbN6yFjIVNwz2vqXOG5tsZ/5AWrctnu7kaanaHvmXgVgIkijHmzW21ZQQANPNgjGBkycGUXZMMUIwGwYJKoZIhvcNAQkUMQ4eDABNAHkAQwBlAHIAdDAjBgkqhkiG9w0BCRUxFgQU59WMkYzN/aRcemeMQJxzpRcC5nwwMTAhMAkGBSsOAwIaBQAEFN17vdeYvqbjC2HcRpRxhBWpv7ydBAgrPrMeGR5G/wICCAA=
```

Comment:
is this valid PKCS 12?
I cant decode it. But i can push it to Azure KV :raised_eyebrow: 

```
echo 👆 | base64 -d > foo
openssl pkcs12 -in foo -info
Enter Import Password:
MAC: sha1, Iteration 2048
MAC length: 20, salt length: 8
PKCS7 Encrypted data: pbeWithSHA1And40BitRC2-CBC, Iteration 2048
Error outputting keys and certificates
40A73646F07F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:373:Global default library context, Algorithm (RC2-40-CBC : 0), Properties ()
```

What also works (which is worth documenting):

```
apiVersion: v1
kind: Secret
metadata:
  name: foobar
stringData:
  certPem: |
    -----BEGIN CERTIFICATE-----
    MIIFEjCCAvqgAwIBAgIUEIP3tmbS5PHz9E2QGCCEBm9ZzMowDQYJKoZIhvcNAQEL
    [...]
    sA+bQNHh
    -----END CERTIFICATE-----
    -----BEGIN PRIVATE KEY-----
    MIIJQwIBADANBgkqhkiG9w0BAQEFAASCCS0wggkpAgEAAoICAQDXLB6fBkzR4lkz
    [...]
    nWmoZasD476N2Sfm0JVeSFch6WEMv5k=
    -----END PRIVATE KEY-----
```

Reply:
I think it is URL-Encoded 😅if you try it with it, it should work

Reply:
I might refactor it to consider b64-encoded, I think it makes sense

Reply:
This isn't URL encoded, there are invalid characters like / or +

Reply:
Ok, just tested here your exact commands and for me they worked. I was asked to provide a password to PEM encrypt the private key, but I believe that was expected on my side (maybe some incompatibility with macs? 😅 )

```
Enter Import Password:
MAC: sha1, Iteration 2048
MAC length: 20, salt length: 8
PKCS7 Encrypted data: pbeWithSHA1And40BitRC2-CBC, Iteration 2048
Certificate bag
Bag Attributes
    friendlyName: MyCert
    localKeyID: E7 D5 8C 91 8C CD FD A4 5C 7A 67 8C 40 9C 73 A5 17 02 E6 7C 
subject=C = AU, ST = Some-State, O = Internet Widgits Pty Ltd

issuer=C = AU, ST = Some-State, O = Internet Widgits Pty Ltd

-----BEGIN CERTIFICATE-----
MIIFazCCA1OgAwIBAgIUZpWjMAnjrJbpE7BODohvgG8KPnMwDQYJKoZIhvcNAQEL
BQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM
GEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMzAxMDkxNDQ4NTNaFw0yNTEw
MjkxNDQ4NTNaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw
HwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggIiMA0GCSqGSIb3DQEB
AQUAA4ICDwAwggIKAoICAQCrggsBYUVC6wHXWZswQJ9EXy9gFXJeNtoCzfIc1Bik
yYJXoq2Un1b7ZMsQbtnJrtektyc1bkCQ3eqRqaxdxtYYrUUf6kmrIn512MqTb1aR
MZu68vG8JUePOvA7sIS7kVlv9ZyqvfeiLFqCno8RpUkYrNLEiFL931YZGfsLYnEk
XJlk8pldwwnwfg1/I/v5nlg3YuJ3zKBVFkJjHcEpm7CUGCnYlzaN3svFl35uXAMY
U8vxPM40p/VA5uC+BVxrzcJVTYqDO0YLhpd1PjA0X5aca+HeRGRU+PTiWeTyAznc
UiAYhaONloRa+h0DRFFcqn+vbRHyFNDnJjRvHPlEBGssZ1w97PghypF2eBGa1VZ0
KhjfozZq04YWw7AUGUmcRdNwapjKCe9cFSxS0wmBdzMJk+6z1LzqYuVyP1j6kyip
FwKLT43SbC6jkrC2aYnRMWlYa0ooRFQk18aflvXMWReWV77LzffYK/o6tiwHNyuQ
kTxsYV475hxiNz4WdGLutpjOk7hDHbLDnUd1Gv7uNarqqTPXdFWuTFEJIRSemZDn
cOgjpxkKMK1HpWC5RwuBvATeXi9idYRbS4zp9zPI/OYpX73cMTBTwLNYY6DLaTrR
euW9sGnr6VZ55WDvYh/oOgnZNVgD4R4k1fSKG5M2MLqLtRVe+ckRxompISrkDrz0
rwIDAQABo1MwUTAdBgNVHQ4EFgQUTMaEn5tN2hHPMQsvw7IwpLa46u8wHwYDVR0j
BBgwFoAUTMaEn5tN2hHPMQsvw7IwpLa46u8wDwYDVR0TAQH/BAUwAwEB/zANBgkq
hkiG9w0BAQsFAAOCAgEABoahgHKwieRwXn//I3NnGKWukWrybmQr5Z2Sby0A5tcS
uV4V/nFNG8TN8G2HH8qbysY2/OAlWmORweiN+V65NcMeISmSatn9ODn07YzSEyJI
ugm0sVxWQFsOaOriQN6V58Mxm1qImWAXtJWuBALGMoE4NItaErXRdO+oZIuXpNhV
yAMAfG06Jya9T/QcDSLnC97I5mb3iR80HuhTPLVLHeEOTc5dDp4pNea2R01dMOYv
fdFpVqB6DTwDbzl78Slxznksq2BF8zBlMQrRMQfP/sZ/7i6q1zqlQbdcrq/rvk/R
HaMRjp9rAhbmIaGKS7WdCYBul6El/RjCQhnUhX39BtED48EByJWcyv/lqiMLwH7/
+keLm4GcM6trmzLltjLfVPU2xRZkYhd/gtwtH41ZClTzDOywITWFkWIAdMyDY7YH
6xmOhU46aBNMNr03SM/eLka2a07BN82y7kLATA4mtfuhLuCHGSxz+bZz3ujzpQOp
HDIChjZMv5YAm+FRcXVJXiol7/RNFhyhSde8wP8i/ciu5NIOHZD8vS/oyiZYqF9f
dBtpaw2M5mIVM5UvFbuZQAnFAKJximCRoU2ADlOPuaKlkBnq7doYHYA2m8tgW6Dd
YivXN/jHAWVLsoXJ1tegv3sYOXVbVeU02gYdABkAqAF3CNF3LEUOlVVSuuLFeSo=
-----END CERTIFICATE-----
PKCS7 Data
Shrouded Keybag: pbeWithSHA1And3-KeyTripleDES-CBC, Iteration 2048
Bag Attributes
    friendlyName: MyCert
    localKeyID: E7 D5 8C 91 8C CD FD A4 5C 7A 67 8C 40 9C 73 A5 17 02 E6 7C 
Key Attributes: <No Attributes>
Enter PEM pass phrase:
Verifying - Enter PEM pass phrase:
-----BEGIN ENCRYPTED PRIVATE KEY-----
MIIJnDBOBgkqhkiG9w0BBQ0wQTApBgkqhkiG9w0BBQwwHAQIoWQjQm0MCR0CAggA
MAwGCCqGSIb3DQIJBQAwFAYIKoZIhvcNAwcECK/3YlAS3R5+BIIJSNq68c0XeUUG
wwF5NuXg3ZK3UNHPa/ggc+6gsaqxqdfUc6DHhaE6MDf4eSlldkrXgfXbVngkoaQs
8vsk3G0nh25vW3SrckQBHnKzaGT94CwbSoZhBy6x/aTclR1nj/hWWoYiiir0Ni9p
180A+fUepCd0F/UIarHyZTEGffUrVqjcBAWDe7huPGn8vtJcxl8huFeiarrlgmlr
fRUVhpOthQHHpuF7ZyIYPXR0nH9YGjAvv9ZIJ7OCP3aRyKjo56ime8Ed4MXZfjoL
jX86abbSYFdk9Pmqn6dn4A/drNFRpOA94LYxThHZvNbsHAhqDzA/7TzQrY48F4oe
tIZhfKRgm/wHl+tK0H2p43yNNknJUmWbsNKC08EPuj+Iyhakp1rH5FwOpsOIcAh3
9dwSznE6yOBPslIqURTAzu6IARhuGdNR9m1jqpLno0QMqt8vXzx7q9sxlCb1F70P
ruoET0bIGCFQxUmVTSnZYr1WfSiv6YXn4FShGV2qJYj5fQUPryiMx+HAJzeT0h2J
Xx3AJzEHAawbFEtFHbyJE7EjObz1J2AK0zhKVPN3lgV59Z5Bw1yu/mzoFDsvLqbx
6bj1rAxYJuarxU8MThPe18jzUuk0syUcFRrYNXuexQPzexCJLm7cB71NOYZpTjhI
kc1g8NXQywxpvgU9syGyZB9YU2IcyLxoGJFqlEvsI/2Z8Q0CTUa9e5ef0rysnvSF
IspNquniLp6ooZ2QemcI7NnI7cQDdT4YmKm+Yx3j+zBiWuTlDvMrcVrz701rY6wk
lNTilEQluIjID3uE5vXUvCNeECpBK6ogvm1s6voATGUGPQzEl6l/I5qxI1Pn7ZB7
IpGXHzXxvekpvTsBM3eqIVyekwg+mKCFDPiW8otkudYQ+jElSqQIqQFnwfUrNMoZ
6NLwvHYFTrDaTFn4GvxP0E9fFLy7wv+pzM5yJtt042LxqIrIhvXwjrGENyPCJw7v
7dvKf0yJu70nJcJP3P4H6wmZT+HHA1IRsvAlL50DxMBmDx/iRE/J7GVL69EvPZ+w
EG5ZqmkGl9vYL4//tOBKvRjgrBOd9aPksa0vIKVbDSAlZBKNwVLLlEwMjvbZRq4D
0n4tnQeVipU9fuhYjT5QLBfYKo7Vcdok6SoWgJjw7NnDS60Ucpbge2ZE4wescpux
5v9QAmoBgriOdWMi3DAseVnKyfjfw+opZI4a8vTCLY9llOVjQoUrOileMWTSnsMh
ly0kwSuWzJJ4tTruyc3tsCbYX5PRyKPxSPq8LKUcOVzSZvLBz8DXTRP/Qy5rBFqJ
DuITOsJuspClB+RV0Rk2F9OKbAqqZ/r+AWIrDoR2RwvYwYwaBBC1qldtkGwUJaIu
s02IUWjyaNZaF+9vJOY1HAbKwqc0tIDhR8+8N40Hwdq0fWtwaDqfTGmxOBtFJG5e
jOLMP4/dJ5UVjNsPSEpzN5tL0rOo1BcwVQYXG/pZ4Zo5omHvhsmTq22PoeJVnoXe
3EV0FA47qWmdXz5QF5YGWzsH4EipOevpZggZNeS/pXuOtmRYitOIdHgMl9PX8UUk
cTr71mC7oCuo7f3sjW72WVMM1nN7OjJ487GVeiZyOfW62OW83Dv6O21OfkBUKoDt
KZ4jFob5B7gAfA03fjbgMsanf6N6Tvodok7msifbIkJOkquA5De2SiOJ+1GRNkGd
o+niBDQqOqMPcIzfwmO5WBeuUFpix3Mms4eo0rVGsV/Ejj764JVsotebWsFPr5Rw
AJhOVxXejYbxkK+fTYN9KJpVDBgoZ+eGtAgAUXbgITqwXCGQECPGj+tgld2MbKKi
n3t2GKX+BPpgO2Ej+hG6Om5d1YEXAswHcAY5EvVpt62IQ9wGCXPFUZX1x6v3i05a
6Alj1Ge7KabgmP+MmyFNC7BQojhLVxjMFXcE8VLVhwipVbndcybOqFSghM1PSJAD
dA166ipyh3FsiS8K5tOUKea8IH1QFtQ4W7mxvVsH62A3T0/tQiEwHFj1lJPn7jFz
K73RNaSqUX5Q8UV2buePd0DWtZCAiHbOqqqV3wISYHtl9hLDR0J+6Jle8tkOtVho
xDGMswPxoYRBEGwWCk1L4pfgNdxepowW6U1D0NfWescvSkhtaA/UhYND5gEJVwiI
LgrA/pgXANY3J3oax2i2a2PYNApa71wv3NKeUuBNPU3vJtm6GIHTp86jlJiXjkqj
/xtiNs2TSYEywyxUPm5ug7POymC//Ab65UDi76+T6aJJZgdUBzHltmajcHBy4N+u
E2Bwl8mjgUzdehcDfnL6P132filZ7VYBR7dvCprmcb33cKMVGUaGU9+tRCFgKURp
bkWTa+K8tLHfHl/H6Z/OhjDDBim44uKeTlE3zJB2eE2gcRdkIF3+9nzQS9lhFI6C
iE0pIgZiEGVtcryMNHlFg3pE5d7KMxoR1KvWpOi9wjFh1go/6oEHT6v0mzLxvC8j
T5CeGVCEBI9noIIclhdTlcnY6FsiiRdxXEYGr4MxV0wOxY8UWreMvN3a5bSTqHRp
YEtzAoxcTaF7rNA3g69tV1lMxN4K9yoFdQyjNpWW5+7eERRuFZIUG9QOpOQ0GGRV
wJmDsb5GYGshnZ9GKwZDWrIbxBtkpv9r8yJPHMBEPd13u+4k+lqTNXg4/Xqy1ZkX
Ny31UJ+7HLN7VLwcxeMSQUQjFvwdm/FFNz8ks4Ejb2LUGlc2om3/tlfS25IXofv5
L468H0l1rGoy4KeEE+E+/NRis9vrlSviBk8X5rRxPkE9HNSmh7MPAevjXnB6PniE
Eg1JIaMe5jWo/9oipZpk3PNyknHqMpu54/EfQsLMtICDCCN56evWm7zVnx4SZ5ji
qD/HJXG8Te/vpkecwUoTh3MtQGBeL8XxrWQCG1+wTJHgTjD9Ll9Y+DluX/EgIOGP
UdraefH3PaWzBETSFONJKxSTSUdAM6iGFYitqA75zU0Hn9eRt0OYsCNW2Z0AMZZ5
nYe09wtO3jI9PD3P43BZfxRQMbMeddTe4lXsiAr6P8/dGNHXZjfpANFgzPMQaXQb
aJzyNVUGUQspCMSdNUuzz2UM+iClWg5fPkNWwmi0VcILCW/Zl2yjy8NMAFkBT2xY
tGE2lAXJ1sMp3QFPVzYCfrJDAdB76PSr7l0zLDwNdsDkMe8pNvsAKQhYtN6i+vkl
RMn8+x4sB4wo4RzFw4scSQ==
-----END ENCRYPTED PRIVATE KEY-----
```

Reply:
:shrug: :shrug:  well then, nevermind :shrug:  :shrug: